### DISCUSSION TITLE
Multi-threading for simulating apparent resistivities/foward modeling?

### DISCUSSION THREAD

**Discussion by benbpete** [gimli-org/pyGIMLi#464] [Ideas]
Hi All,

I'm not well versed in how computers use their resources, but I did see in the C++ core that pygimli implements multi-threading when generating the Jacobian. This really seems to speed up the inversion process.  Is there anyway to implement multiple threads when simulating apparent resistivities?  I'm not sure if there are any computations that would benefit from it, but I thought I would bring it up.

Thanks!

Ben

**Comment by carsten-forty2** [comment]
Hi Ben,

thanks for pointing out and your are right. Not all possible benefits from parallelisation are used at the moment.

There are several topics on my TODO regarding performance for the ERT process. 
I would expect a little speedup using multithreading while solving the equation system for multiple sources. Also the matrix assembling could be maybe a little faster in parallel. However, there are other parts in the ERT process to be reviewed with higher priority, which also might have a larger income in the sense of performance boost. But its all a question of time and prioritry ;)

Cheers,
Carsten

**Reply by benbpete** [reply]
Thanks for the response Carsten.  I'm sure there is a lot to review.  Thanks for all of the work you put into pygimli!
